packages/contracts-periphery/contracts/foundry-tests/Optimist.t.sol

```diff
@@ -284,54 +488,142 @@ contract OptimistTest is Optimist_Initializer {
 
         // expect bob to have no balance now
         assertEq(optimist.balanceOf(bob), 0);
     }
 
     /**
-     * @dev setApprovalForAll should revert as sbt
+     * @notice Non-owner attempting to burn token should revert.
      */
-    function test_optimist_set_approval_for_all() external {
-        attestAllowlist(bob);
+    function test_burn_byNonOwner_reverts() external {
+        _mockAllowlistTrueFor(bob);
 
         // mint as bob
         vm.prank(bob);
         optimist.mint(bob);
-        vm.prank(alice_admin);
-        vm.expectRevert(bytes("Optimist: soul bound token"));
-        optimist.setApprovalForAll(alice_admin, true);
 
-        // expect approval amount to stil be 0
-        assertEq(optimist.getApproved(256), address(0));
-        // isApprovedForAll should return false
-        assertEq(optimist.isApprovedForAll(alice_admin, alice_admin), false);
+        vm.expectRevert("ERC721: caller is not token owner nor approved");
+        // burn as Sally
+        vm.prank(sally);
+        optimist.burn(_getTokenId(bob));
+
+        // expect bob to have still have the token
+        assertEq(optimist.balanceOf(bob), 1);
     }
 
     /**
-     * @dev should support erc721 interface
+     * @notice Should support ERC-721 interface.
      */
-    function test_optimist_supports_interface() external {
+    function test_supportsInterface_returnsCorrectInterfaceForERC721_succeeds() external {
         bytes4 iface721 = type(IERC721).interfaceId;
-        // check that it supports erc721 interface
+        // check that it supports ERC-721 interface
         assertEq(optimist.supportsInterface(iface721), true);
     }
+
+    /**
+     * @notice Checking that multi-call using the invite & claim flow works correctly, since the
+     *         frontend will be making multicalls to improve UX. The OptimistInviter.claimInvite
+     *         and Optimist.mint will be batched
+     */
+    function test_multicall_batchingClaimAndMint_succeeds() external {
+        uint256 inviterPrivateKey = 0xbeefbeef;
+        address inviter = vm.addr(inviterPrivateKey);
+
+        address[] memory addresses = new address[](1);
+        addresses[0] = inviter;
+
+        vm.prank(eve_inviteGranter);
+
+        // grant invites to Inviter;
+        optimistInviter.setInviteCounts(addresses, 3);
+
+        // issue a new invite
+        OptimistInviter.ClaimableInvite memory claimableInvite = optimistInviterHelper
+            .getClaimableInviteWithNewNonce(inviter);
+
+        // EIP-712 sign with Inviter's private key
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            inviterPrivateKey,
+            optimistInviterHelper.getDigest(claimableInvite)
+        );
+        bytes memory signature = abi.encodePacked(r, s, v);
+
+        bytes32 hashedCommit = keccak256(abi.encode(bob, signature));
+
+        // commit the invite
+        vm.prank(bob);
+        optimistInviter.commitInvite(hashedCommit);
```

Comment:
Signature generation isn't deterministic so including the sig in the commit hash is slightly sus

Reply:
Discussed with Mark:

- The secp256k1 signature generation is non-deterministic, so if the issuer attempts to create the same signature twice using the same payload, they may end up with two different signatures
- This shouldn't be a big problem in the intended use for this contract, as it's a very unlikely scenario where the issuer attempts to create the same signature twice. 

Signature issuer side
- They issue a signature, then it gets stored in localStorage
- They create a url with the signature in the params that they send to the invitee

Recipient side
- They receive the link through chat or messenger
- They go to the link, and the signature gets stored in their localStorage
- They commit the signature
- They reveal with the signature in their local storage

If the recipient’s browser cache is lost, they need to find the link again in order to continue with the process.

At worst, the if the recipient commits an invite, then loses the link, and clears the browser cache, they need to ask the issuer to send them the link again. If the issuer has also lost the local copy of the created signature, it is difficult for the issuer to generate the same signature. In this case, it’s recommended that the issuer just generates a new signature/link, and the recipient goes through the commit/reveal process again.